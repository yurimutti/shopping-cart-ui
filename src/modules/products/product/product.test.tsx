import { render, screen, fireEvent } from "@testing-library/react";
import { describe, it, expect, vi } from "vitest";
import { Product } from "./";
import type { Product as ProductType } from "@/types/product";

const mockProduct: ProductType = {
  id: 1,
  name: "Cheeseburger Combo",
  price: 12.99,
  quantity: 1,
};

describe("<Product />", () => {
  it("should render product information correctly", () => {
    const { container } = render(
      <Product product={mockProduct} onAddToCart={vi.fn()} />
    );

    expect(screen.getByText(mockProduct.name)).toBeInTheDocument();
    expect(
      screen.getByText(`$${mockProduct.price.toFixed(2)}`)
    ).toBeInTheDocument();

    expect(
      screen.getByRole("button", {
        name: `Add ${mockProduct.name} to cart`,
      })
    ).toBeInTheDocument();

    expect(container.firstChild).toMatchSnapshot();
  });

  it("should call onAddToCart when add button is clicked", () => {
    const onAddToCart = vi.fn();

    render(<Product product={mockProduct} onAddToCart={onAddToCart} />);

    const addButton = screen.getByRole("button", {
      name: `Add ${mockProduct.name} to cart`,
    });

    fireEvent.click(addButton);

    expect(onAddToCart).toHaveBeenCalledWith(mockProduct);
  });
});
