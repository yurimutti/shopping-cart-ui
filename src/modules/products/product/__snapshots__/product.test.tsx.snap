// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<Product /> > should render product information correctly 1`] = `
<div
  class="rounded-2xl bg-white p-6 flex flex-col items-center shadow-md border border-gray-200"
>
  <h2
    class="text-lg font-semibold mb-2 text-black text-center"
  >
    Cheeseburger Combo
  </h2>
  <p
    class="text-gray-700 mb-4"
  >
    $
    12.99
  </p>
  <button
    aria-label="Add Cheeseburger Combo to cart"
    class="rounded-full bg-red-600 hover:bg-red-700 text-white px-4 py-2 text-sm font-medium transition-colors"
  >
    Add to Cart
  </button>
</div>
`;
